### PR TITLE
Fix exports for files in lib

### PR DESCRIPTION
**Pull request by LKay**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Original module uses `babel-add-module-exports` plugin which adds `module.exports = exports['default']` line at the end of each transpiled file and thus breaks default import in typescript because module is converted into common js module. This addresses that and makes definitions of files inside `lib` be generated in common js style.

### CODE REVIEW

**Comment by andy-ms** [comment]
It looks like the latest version of `react-icons` isn't transpiled? If I run `npm install react-icons` and navigate to `node_modules/react-icons/fa/500px.js`, I see `export default Fa500px`.

**Comment by LKay** [comment]
I noticed that too. However this change applies to everything in `lib` directory, which is transpiled.

**Comment by andy-ms** [comment]
Ah, OK. Thanks!

**Comment by johnnyreilly** [comment]
Oh dear - this has broken me.  With this version of the changes the following code:

```
import Fa500px from 'react-icons/lib/fa/500px';
```

errors out with:

> [ts] Module '"c:/work/treasury-tools/src/Treasury.App/node_modules/@types/react-icons/lib/fa/500px"' has no default export.

Given the following TypeScript definition:

```
import * as React from 'react';
import { IconBaseProps } from 'react-icon-base';
declare class Fa500px extends React.Component<IconBaseProps> { }
export = Fa500px;
```

The previous `export default ....` worked fine.

Is there a way that we can fix this that suits both @LKay and myself?  For context; here's the actual JS that is being typed in this case:

```
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactIconBase = require('react-icon-base');

var _reactIconBase2 = _interopRequireDefault(_reactIconBase);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Fa500px = function Fa500px(props) {
    return _react2.default.createElement(
        _reactIconBase2.default,
        _extends({ viewBox: '0 0 40 40' }, props),
        _react2.default.createElement(
            'g',
            null,
            _react2.default.createElement('path', { d: 'm35.3 34.5l-0.2 0.2q-2.5 2.5-5.7 3.9-3.5 1.4-7.1 1.4-3.7 0-7.1-1.4-3.3-1.4-5.8-3.9-2.5-2.5-3.9-5.8-0.9-2.3-1.2-4.2-0.1-0.6 1.1-0.8 1.1-0.2 1.2 0.4 0.1 0.1 0.1 0.1 0.4 2 1 3.6 1.1 2.7 3.4 5 2.2 2.2 5 3.4 3 1.3 6.2 1.3 3.2 0 6.1-1.3 2.9-1.2 5.1-3.4l0.1-0.1q0.2-0.2 0.6-0.2 0.2 0.1 0.7 0.5 0.8 0.9 0.4 1.3z m-10.6-13.7l-1.4 1.5 1.4 1.4q0.4 0.5-0.2 1.1-0.4 0.4-0.7 0.4-0.2 0-0.4-0.3l-1.4-1.3-1.5 1.4q-0.1 0.2-0.3 0.2-0.4 0-0.7-0.4l-0.1 0q-0.4-0.4-0.4-0.7 0-0.1 0.2-0.4l1.5-1.4-1.5-1.5q-0.3-0.4 0.3-1 0.4-0.4 0.7-0.4 0.2 0 0.3 0.1l1.5 1.5 1.4-1.5q0.4-0.3 1.1 0.3 0.6 0.6 0.2 1z m10.6 1.3q0 2.6-1.1 5.1-1 2.3-2.8 4.1-1.8 1.8-4.2 2.8t-5.1 1.1-5-1.1-4.2-2.8q-1.8-1.8-2.8-4.1-0.3-0.7-0.3-0.9h-0.1q-0.2-0.6 1-1 1.1-0.4 1.3 0.3 0.9 2.2 2.2 3.7h0v-7.6q0.1-3.1 2.3-5.2 2.3-2.3 5.6-2.3 3.3 0 5.7 2.3t2.3 5.6q0 3.2-2.4 5.6t-5.6 2.3q-1.2 0-2.5-0.4-0.6-0.2-0.2-1.3 0.3-1.2 0.9-1l0.4 0.1q0.3 0 0.7 0.1t0.7 0.1q2.3 0 3.9-1.6t1.6-3.9q0-2.3-1.6-3.8-1.6-1.6-3.9-1.6-2.4 0-4 1.8-1.4 1.6-1.4 3.5v9.2q2.4 1.5 5.4 1.5 2.1 0 4.1-0.8t3.5-2.3 2.3-3.4 0.8-4.1q0-4.4-3.1-7.6-3.2-3.1-7.6-3.1-4.5 0-7.6 3.1-1.2 1.2-1.7 1.9l0 0.1q-0.2 0.2-0.3 0.3t-0.5 0.3-0.9-0.1q-0.4-0.1-0.8-0.4t-0.3-0.6v-15.2q0-0.3 0.2-0.5t0.6-0.3h19.6q0.7 0 0.7 1.2t-0.7 1.3h-18.1v10.7h0q0.9-0.9 2.3-1.8t2.4-1.4q2.4-1 5.1-1 2.7 0 5.1 1t4.2 2.8q1.8 1.8 2.8 4.2 1 2.5 1 5.1z m-0.7-13q0.2 0.2 0.2 0.4t-0.2 0.4-0.3 0.5q-0.6 0.6-0.9 0.6-0.2 0-0.4-0.2-2.3-2-4.6-3-2.8-1.2-6.1-1.2-3 0-5.9 1.1-0.6 0.2-1-0.8-0.2-0.6-0.2-0.9 0.1-0.4 0.4-0.4 2.9-1.3 6.7-1.3 3.6 0 7 1.4 3.1 1.3 5.3 3.4z' })
        )
    );
};

exports.default = Fa500px;
module.exports = exports['default'];
```

**Comment by johnnyreilly** [comment]
@andy-ms I'd appreciate your guidance if possible...

**Comment by LKay** [comment]
The very last line exposes module as commonjs module and this should be defined and used as such. So you would have to do `import Fa500px = require(‘react-icons/lib/fa/500px’)` to follow the code.

**Comment by johnnyreilly** [comment]
Alas whilst that compiles it doesn't work.  To take an example, the following code:

```
import FaSignIn = require('react-icons/lib/fa/sign-in');
```

Compiles happily, but when you run it dies with:

```
ReferenceError: FaSignIn is not defined
    at Header.webpackJsonp../src/components/layout/header.tsx.Header.render (header.tsx:88)
    at Object.allowStateChanges (mobx.module.js:969)
    at index.module.js:771
    at trackDerivedFunction (mobx.module.js:2908)
    at Reaction../node_modules/mobx/lib/mobx.module.js.Reaction.track (mobx.module.js:3076)
    at reactiveRender (index.module.js:766)
    at Header.initialRender (index.module.js:759)
    at ReactCompositeComponent.js:793
    at measureLifeCyclePerf (ReactCompositeComponent.js:73)
    at ReactCompositeComponentWrapper._renderValidatedComponentWithoutOwnerOrContext (ReactCompositeComponent.js:792)
```

**Comment by andy-ms** [comment]
@johnnyreilly `"allowSyntheticDefaultImports": true,`.
Don't know why `import =` is giving you not-defined errors as such an import should at worst give you an object with a `default` property. I guess your module loader might not work at all without default imports.

**Comment by johnnyreilly** [comment]
@andy-ms I'll try flipping the switch and see what `allowSyntheticDefaultImports: true` works. I'm using webpack as my loader so it's probably a fairly common use case. It'll be tomorrow before I can give this a go.

On a different tack I've found myself making use of your ts-loader PR today: https://github.com/TypeStrong/ts-loader/pull/468

It was never merged (mostly down to laziness on my part) but I've been using it as I take a look at another ts-loader issue: https://github.com/TypeStrong/ts-loader/issues/657

Long story short: the module resolution stuff from compiler.sys is likely to move into ts-loader pretty soon. Thanks for the initial PR!

**Comment by johnnyreilly** [comment]
`allowSyntheticDefaultImports: true` looks good!
